api/src/main/java/xtdb/query/Expr.java

```diff
@@ -322,6 +345,81 @@ static Exists exists(Query query, List<ArgSpec> args) {
         return new Exists(query, args);
     }
 
+    final class Vec implements Expr {
+        public final List<Expr> elements;
+
+        private Vec(List<Expr> elements) {
+            this.elements = elements;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Vec vec = (Vec) o;
+            return Objects.equals(elements, vec.elements);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(elements);
+        }
+    }
+
+    static Vec vec(List<Expr> elements){
+       return new Vec(elements);
+    }
+
+    final class Set implements Expr {
+        public final java.util.Set<Expr> elements;
```

Comment:
at the risk of some inconsistency, I'd be tempted to call these three classes SetExpr so that you don't need the FQ java.util.Set